lib/puppet/provider/user/windows_adsi.rb

```diff
@@ -95,8 +269,17 @@ def delete
     @deleted = true
   end
 
-  # Only flush if we created or modified a user, not deleted
   def flush
+    return unless @sync_password
+
+    # The reason we need to sync the password here is because of the
+    # password_change_required attribute. We want to sync. the password
+    # based on the _final_ state of this attribute, which is determined
+    # after the attributes property is synced.
+    debug _("Setting the password ...")
+    set_password(@resource[:password])
```

Comment:
We'll still need to set the password here in `flush` even if we don't need to print the warning for the `password_change_required` attribute because we'll need to print warning messages for other attributes like the `disabled` attribute -- see e.g. https://github.com/caseywilliams/puppet/commit/5b1ec4339dc1e08c98385f1c9071ac7a0884584f#diff-cb6db5b98229b4ec070f9158795e3708R130